src/node-capnp/capnp.cc

```diff
@@ -246,9 +396,9 @@ static constexpr uint NEW_FD_FLAGS =
 
 class OwnedFileDescriptor {
 public:
-  OwnedFileDescriptor(uv_loop_t* loop, int fd, uint flags)
+  OwnedFileDescriptor(uv_loop_t* loop, SOCKET fd, uint flags)
       : uvLoop(loop), fd(applyFlags(fd, flags)), flags(flags),
-        uvPoller(uv_poll_init, uvLoop, fd) {
+        uvPoller(uv_poll_init_socket, uvLoop, fd) {
```

Comment:
This seems to be a substantive change from the original that is not win32 only - a comment explaining the reason for the divergence from upstream would be useful, if not putting this in its own commit.

Reply:
Sorry, I'm confused - this is already in its own commit, 80ebee24ce0db73e3b06440d68325e3f7b43259a

(Looks like the commit message is back-to-front! I'll reword it now.)

Reply:
Ah, I had missed that it was in its own commit. I suppose my point is that given the number of commits in this branch currently, I'm expecting that we would squash it into main, at which point we would lose the reasoning for this change.

That to one side, on Linux, this change causes the event loop to hang as soon as a capability is dispatched.

Reply:
The plan is to rebase